src/gui/advancedsettings.cpp

```diff
@@ -184,6 +192,45 @@ void AdvancedSettings::updateCacheSpinSuffix(int value)
         spin_cache.setSuffix(tr(" MiB"));
 }
 
+void AdvancedSettings::updateInterfaceAddressCombo(int) {
+    //Try to get the currently selected interface name
+    QString ifaceName;
+    if (combo_iface.currentIndex() == 0) {
+        ifaceName = QString();
+    }
+    else {
+        ifaceName = combo_iface.itemData(combo_iface.currentIndex()).toString();
+    }
+    const QNetworkInterface iface = QNetworkInterface::interfaceFromName(ifaceName);
+
+    //Clear all items and reinsert them, default to all
+    combo_iface_address.clear();
+    combo_iface_address.addItem(tr("All Addresses"));
+    combo_iface_address.setCurrentIndex(0);
+    if (!iface.isValid()) {
+        return;
+    }
+    //Found a valid interface, try to get the addresses
+    const QList<QNetworkAddressEntry> addresses = iface.addressEntries();
+    const Preferences* const pref = Preferences::instance();
+    const QString currentAddress = pref->getNetworkInterfaceAddress();
+
+    foreach (const QNetworkAddressEntry &entry, addresses) {
+        QHostAddress ip = entry.ip();
+        QString ipString = ip.toString();
+        QAbstractSocket::NetworkLayerProtocol protocol = ip.protocol();
+        Q_ASSERT(protocol == QAbstractSocket::IPv4Protocol || protocol == QAbstractSocket::IPv6Protocol);
+        //Only take ipv4 for now?
+        if (protocol != QAbstractSocket::IPv4Protocol)
+            continue;
+        combo_iface_address.addItem( ipString );
+        //Try to select the last added one
+        if (ipString == currentAddress) {
+            combo_iface_address.setCurrentIndex(combo_iface_address.count() - 1);
+        }
+    }
+}
```

Comment:
Also for "Any" interface you should populate the combobox with ` QNetworkInterface::allAddresses()`.